core/src/main/java/com/scalar/db/storage/cassandra/CassandraAdmin.java

```diff
@@ -542,6 +542,23 @@ public String toString() {
 
   @Override
   public void upgrade(Map<String, String> options) throws ExecutionException {
-    throw new UnsupportedOperationException("Not yet implemented");
+    createMetadataKeyspaceIfNotExists();
+    createKeyspacesTableIfNotExists();
+    // Retrieve user keyspace and filter out system ones. A downside is that this may include
```

Comment:
Just a question.
Are we going to check the tables of a namespace to see if the namespace is created for Scalar DB?

Reply:
If the table has transaction metadata columns, we can infer it is from Scalar DB. Otherwise, it is either a Scalar DB storage mode table or not a Scalar DB table.
So even if we check each table of the namespace, we can't be sure the namespace is not for Scalar DB.
Also, if a namespace not for Scalar DB is being added to the list of managed Scalar DB namespaces, it causes no harm besides being a false positive.